backend/groth16/bls12-377/prove.go

```diff
@@ -61,35 +63,32 @@ func Prove(r1cs *cs.R1CS, pk *ProvingKey, fullWitness witness.Witness, opts ...b
 
 	log := logger.Logger().With().Str("curve", r1cs.CurveID().String()).Int("nbConstraints", r1cs.GetNbConstraints()).Str("backend", "groth16").Logger()
 
-	proof := &Proof{}
+	proof := &Proof{Commitments: make([]curve.G1Affine, len(r1cs.CommitmentInfo))}
 
 	solverOpts := opt.SolverOpts[:len(opt.SolverOpts):len(opt.SolverOpts)]
 
-	if r1cs.GetNbCommitments() != 0 {
-		solverOpts = append(solverOpts, solver.OverrideHint(r1cs.CommitmentInfo[0].HintID, func(_ *big.Int, in []*big.Int, out []*big.Int) error {
-			// Perf-TODO: Converting these values to big.Int and back may be a performance bottleneck.
-			// If that is the case, figure out a way to feed the solution vector into this function
-			if len(in) != r1cs.CommitmentInfo[0].NbCommitted() { // TODO: Remove
-				return fmt.Errorf("unexpected number of committed variables")
-			}
-			values := make([]fr.Element, r1cs.CommitmentInfo[0].NbPrivateCommitted)
-			nbPublicCommitted := len(in) - len(values)
-			inPrivate := in[nbPublicCommitted:]
-			for i, inI := range inPrivate {
-				values[i].SetBigInt(inI)
-			}
+	commitmentSubIndexes := r1cs.CommitmentInfo.CommitmentIndexesInCommittedLists()
```

Comment:
maybe `CommitmentIndexesInCommittedLists` -> `Indexes` ?

Reply:
I think that wouldn't be expressive enough. This is the list of commitments that are committed to, and they are referred to not even by their wire index, but by their index in the list of committed values in that commitment that commits to them. 🤪